tests/driver_sx1280/sx1280_rangetest_cmd.c

```diff
@@ -0,0 +1,313 @@
+        }
+
+        printf("frm_payload=");
+        printf_ba(frm_payload, frm_payload_len);
+        printf("\n");
+
+        // build the LoRaWAN frame
+        build_lorawan_frame(frame, payload_len, MHDR_DATAUP_CNF, devaddr, fcnt,
+                            FCTRL_NONE, FPORT, frm_payload, frm_payload_len,
+                            NWKSKEY, APPSKEY);
+
+        // Set the IQ to normal for uplink
+        sx1280_invertiq_set(false);
+
+        // prepare the LoRaWAN frame
+#ifdef CPU_ESP32
+        printf("sending : channel=%u iq=%d fcnt=%u frame=", channel, params.invert_iq_is_on,
+               fcnt);
+#else
+        printf("sending : channel=%lu iq=%d fcnt=%lu frame=", channel, params.invert_iq_is_on,
+               fcnt);
+#endif
+
+        printf_ba(frame, payload_len);
+        printf("\n");
+
+        params.pld_len_in_bytes = payload_len;
+        ral_status_t res = ral_sx1280_setup_tx_lora(&ral_default_cfg, &params);
+        if (res != RAL_STATUS_OK) {
+            printf("ral_sx1280_setup_tx_lora ERROR %d\n", res);
+            reboot_after_delay(10);
+            return -1;
+        }
+        else {
+            ral_sx1280_set_pkt_payload(&ral_default_cfg, (uint8_t *)frame,
+                                       payload_len);
+            res = ral_sx1280_set_tx(&ral_default_cfg);
+            if (res != RAL_STATUS_OK) {
+                printf("ral_sx1280_set_tx ERROR %d\n", res);
+                reboot_after_delay(10);
+                return -1;
+            }
+
+            xtimer_sleep(RECEIVE_WINDOWS_IN_SECONDS);
+
+            // Set the IQ to inverted for downlink
+            sx1280_invertiq_set(true);
+
+            // listen downlink on receive windows
+#ifdef CPU_ESP32
+            printf("listen : channel=%u iq=%d duration=%d seconds ...\n", channel,
+                   params.invert_iq_is_on, RECEIVE_WINDOWS_TIMEOUT_IN_SECONDS);
+#else
+            printf("listen : channel=%lu iq=%d duration=%d seconds ...\n", channel,
+                   params.invert_iq_is_on, RECEIVE_WINDOWS_TIMEOUT_IN_SECONDS);
+#endif
+
+
+            params.pld_len_in_bytes = 255;
+            ral_status_t res = ral_sx1280_setup_rx_lora(&ral_default_cfg, &params);
+            if (res != RAL_STATUS_OK) {
+                printf("ral_sx1280_setup_rx_lora ERROR %d\n", res);
+            }
+            else {
+                res =
+                    ral_sx1280_set_rx(&ral_default_cfg, RECEIVE_WINDOWS_TIMEOUT_IN_SECONDS * 1000);
+                if (res != RAL_STATUS_OK) {
+                    printf("ral_sx1280_set_rx ERROR %d\n", res);
+                }
+            }
+
+            xtimer_sleep(RECEIVE_WINDOWS_TIMEOUT_IN_SECONDS);
+
+            // sleep before next frame sending
+            if (i < count) {
+#ifdef CPU_ESP32
+                printf("sleep %d seconds before next frame\n", delay);
+#else
+                printf("sleep %ld seconds before next frame\n", delay);
+#endif
```

Comment:
I also noticed another mistake during my first look.

You don't need specific for ESP32, just use `PRIu32`:

```suggestion
                printf("sleep %" PRIu32 " seconds before next frame\n", delay);
```